modules/serverless-nodejs-functions-context-objects.adoc

```diff
@@ -53,3 +53,64 @@ function handle(context, data)
 ----
 
 The `data` parameter in this example is a JavaScript object that contains the `customerId` and `productId` properties.
+
+[id="serverless-nodejs-functions-context-objects-arbitrary-data_{context}"]
+== Arbitrary data
+
+A function can receive any data, not just CloudEvents. For example, you might want to call a function using POST with an arbitrary object in the body:
+
+[source,json]
+----
+{
+  "id": "12345",
+  "contact": {
+    "title": "Mr.",
+    "firstname": "John",
+    "lastname": "Smith"
+  }
+}
+----
+
+In this case, you can define the function as follows:
+
+[source,javascript]
+----
+function handle(context, customer) {
+  return "Hello " + customer.contact.title + " " + customer.contact.lastname;
+}
+----
+
+Supplying the contact object to the function would then return the following:
+
+[source,text]
+----
+Hello Mr. Smith
+----
+
+[id="serverless-nodejs-functions-context-objects-supported-data-types_{context}"]
+== Supported data types
+
+CloudEvents can contain various data types, including JSON, XML, plain text, and binary data. These data types are provided to the function in their respective formats:
+
+* *JSON Data*: Provided as a JavaScript object.
+* *XML Data*: Provided as an XML document.
+* *Plain Text*: Provided as a string.
+* *Binary Data*: Provided as a Buffer object.
+
+[id="serverless-nodejs-functions-context-objects-handling-different-data-types_{context}"]
+== Handling different data types
+
+Ensure your function can handle different data types by checking the Content-Type header and parsing the data accordingly. For example:
```

Comment:
This line says "procedure" to me, but there's no specific action to take, right?